src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -251,7 +251,7 @@ private bool StartCore(ProcessStartInfo startInfo)
             {
                 filename = ResolvePath(startInfo.FileName);
                 argv = ParseArgv(startInfo);
-                if (Directory.Exists(startInfo.FileName))
+                if (Directory.Exists(filename))
```

Comment:
I think this is the only line that is needed for the fix?

BTW, in the PR description, it is good to describe what caused the bug, and how the change fixes it. It can usually be brief. For this one, you can manually add it when you hit squash to commit. I this case it would be something like,

> We do not support Process.Start on a directory on Unix unless UseShellExecute is set. (When UseShellExecute is set, we can call xdg-open to open a window showing the directory.) When we do this check, we should be testing the actual path we are going to give to execve. When Process.Start is passed the directory (via ProcessStartInfo.FileName) we will attempt to convert any relative path to a fully qualified path. We do this by looking in the directory of the executable, and in the $PATH. We were inadvertently checking the path before this  conversion was done. The effect was that if a file name was passed "X" contained within a directory "\~/X" we would error that "X" was a directory, even though we were going to call execve on "~/X/X". The fix is to make the check after converting the path.


Although that's probably more long winded than necessary.

Reply:
thanks for this summary.

I update the secondary commit with the above content:
https://github.com/dotnet/corefx/pull/24498/commits/8f68e17a5405399e7f18b7c7063c8cd0bd4c3f72

As you suggested, I'm going to use this message when doing squash and merge